clang/lib/Sema/Sema.cpp

```diff
@@ -668,13 +669,15 @@ bool Sema::ImpCastPointerToCHERICapability(QualType FromTy, QualType ToTy,
   assert(ToTy->isPointerType() && "Target types should be PointerType");
   assert(ToTy->getAs<PointerType>()->isCHERICapability() &&
          "Target type must be a capability pointer");
+  // We should not be performing these checks for dependent or placeholder
+  // types (e.g. overloaded function references) since they will be done again
+  // once the actual type has been resolved.
+  assert(!(FromTy->isDependentType() || FromTy->isPlaceholderType()) &&
```

Comment:
What about ToTy?

Reply:
Yes it should probably also apply to `ToTy`, I'll add the assert and see if it breaks anything.

Reply:
Probably worth putting it as the first assert in the function, too, as I don't know if any of the earlier ToTy ones will explode in interesting ways if ToTy is dependent.